document/AnnexG-Examples.mdk

```diff
@@ -0,0 +1,466 @@
+# Annex G (informative) Examples { - #Annex-G }
+
+This Annex contains examples of complete, valid SARIF files,
+to complement the fragments shown in examples throughout this document.
+
+## Minimal valid SARIF file resulting from a scan { - #AnnexG-Minimal }
+
+This is a minimal valid SARIF file for the case where the analysis tool was run with the
+intent of scanning files and producing results (see &sect;[#run-results]).
+The file contains only those elements required by the specification
+(that is, those elements which the specification states "shall" be present).
+
+The file contains a single `run` object (&sect;[#run]) with an empty `results` array (&sect;[#run-results]),
+as would happen if the tool detected no issues in any of the files it scanned.
+
+```
+{
+  "version": "1.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "semanticVersion": "2.1.0"
+      },
+      "results": [
+      ]
+    }
+  ]
+}
+```
+
+## Minimal recommended SARIF file resulting from a scan { - #AnnexG-Recommended }
+
+This is a minimal recommended SARIF file for the case where the analysis tool was run with the
+intent of scanning files and producing results (see &sect;[#run-results]).
+The file contains those elements recommended by the specification
+(that is, those elements which the specification states "should" be present),
+in addition to the required elements.
+
+The file contains a single `run` object (&sect;[#run]) with a `results` array (&sect;[#run-results]).
+The `results` array contains a single `result` object (&sect;[#result]) 
+so the recommended elements of the `result` object can be shown.
+
+It contains a `files` object (&sect;[#run-files]) specifying only those files in which the tool detected a result.
+It contains a `logicalLocations` object (&sect;[#run-logicalLocations]) specifying only those logical locations in which the tool detected a result.
+
+This example also includes rule metadata (&sect;[#run-rules]), even though rule metadata is optional,
+to show how a SARIF log file can be self-contained, in the sense of containing all the information
+necessary to interpret the results.
+
+```
+{
+  "version": "1.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "semanticVersion": "2.1.0"
+      },
+      "files": {
+        "file:///user/builder/work/src/collections/list.cpp": {
+          "mimeType": "text/x-c"
+        }
+      },
+      "logicalLocations": {
```

Comment:
ok, but then I'm also going to clarify the spec about when `logicalLocations` "should" be present and when it only "may" be present. As the spec stands, it unequivocally states that `logicalLocations` "should" be present.

---
In reply to: [72475130](https://github.com/sarif-standard/sarif-spec/pull/248#discussion_r72475130) [](ancestors = 72475130)